### PR TITLE
React/DP-12293: Currency component.

### PR DESCRIPTION
**Pull request by smurrayatwork**
## Description
This adds the Input and InputCurrency components to mayflower react. This PR adds the [Numbro library](https://numbrojs.com/) for support currencies.

Both Input and InputCurrency are exported here, as well as Input's context, so that anybody using mayflower react could potentially use Input for their own input components.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-12293)
- [Github issue]()

## Steps to Test
1. Pull the branch and run  `npm install`, then start up storybook. Browse to http://localhost:6006 then go to Atoms -> forms -> InputCurrency. Verify there's no errors and that all knob settings work.

## Screenshots
screencapture-localhost-6006-2019-01-24-10_57_18

### CODE REVIEW

File: changelogs/DP-12293.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+___DESCRIPTION___
+Added
+Minor
+- (React) DP-12293: Adds InputCurrency component.
```
Comment (avrilpearl) [inline changelogs/DP-12293.txt]: do you want to add a note here about the numbro and input component as well.

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  /** Min value for the field. */
+  min: PropTypes.number,
+  /** Using the up/down arrow keys will increment/decrement the input value by this number. */
+  step: PropTypes.number,
+  /** A language tag that represents what country the currency should display. Comes from IETF BCP 47: https://numbrojs.com/languages.html */
+  language: PropTypes.string,
+  /** Numbro Formatting options for displaying the currency. See https://numbrojs.com/format.html */
+  format: PropTypes.object
+};
+InputCurrency.defaultProps = {
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js:280]: can you add some more defaults to this:
- step = 0.01 
- language = 'English'
 - format: {
    mantissa: 2,
    trimMantissa: false,
    thousandSeparated: true,
    negative: 'parenthesis'
  }

File: react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  width: () => number('InputCurrency.width', 0),
+  maxlength: () => number('InputCurrency.maxlength', 20),
+  placeholder: () => text('InputCurrency.placeholder', 'type something'),
+  errorMsg: () => text('InputCurrency.errorMsg', 'you did not type something'),
+  defaultText: () => text('InputCurrency.defaultText', '0'),
+  max: () => number('InputCurrency.max', 1000),
+  min: () => number('InputCurrency.min', -1000),
+  step: () => number('InputCurrency.step', 0.01),
+  format: () => object('InputCurrency.format', {
+    mantissa: 0,
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js]: can you change this to 2 for the story

File: react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  labelText: () => text('InputCurrency.labelText', 'Text Input'),
+  required: () => boolean('InputCurrency.required', false),
+  id: () => text('InputCurrency.id', 'text-input'),
+  name: () => text('InputCurrency.name', 'text-input'),
+  width: () => number('InputCurrency.width', 0),
+  maxlength: () => number('InputCurrency.maxlength', 20),
+  placeholder: () => text('InputCurrency.placeholder', 'type something'),
+  errorMsg: () => text('InputCurrency.errorMsg', 'you did not type something'),
+  defaultText: () => text('InputCurrency.defaultText', '0'),
+  max: () => number('InputCurrency.max', 1000),
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js]: can we increas this to 10,000 for the story so users can see the comma sep formatting

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                const langMax = langKeys.length;
+                for (; i < langMax; i += 1) {
+                  const langKey = langKeys[i];
+                  const lang = languages[langKey];
+                  numbro.registerLanguage(lang);
+                }
+                numbro.setLanguage(props.language);
+              }
+              const currency = numbro(number);
+              const format = props.format || {
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]: should we move this into default props

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                update.errorMsg = errorMsg;
+              }
+            } else {
+              errorMsg = '';
+              update.showError = false;
+              update.errorMsg = errorMsg;
+            }
+            context.updateState(update);
+            if (typeof props.onChange === 'function') {
+              props.onChange(numberValue);
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]: can you pass the event to this function as well.

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+import Input from '../Input';
+import { InputContext } from '../Input/context';
+
+const Currency = (props) => (
+  <React.Fragment>
+    <InputContext.Consumer>
+      {
+        (context) => {
+          const inputClasses = classNames({
+            ma__input: true,
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputCurrency/index.js]: can we add a specific currency class for this input.
```suggestion
            'ma__input-currency': true,
```

File: changelogs/DP-12293.txt (hunk 1 above)
Comment (smurrayatwork) [inline changelogs/DP-12293.txt]:
```suggestion
- (React) DP-12293: Adds Input, InputCurrency components and InputContext context. Uses Numbro.js library.
```

File: react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js (hunk 4 above)
Comment (smurrayatwork) [inline react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js]:
```suggestion
  max: () => number('InputCurrency.max', 10000),
```

File: react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  width: () => number('InputCurrency.width', 0),
+  maxlength: () => number('InputCurrency.maxlength', 20),
+  placeholder: () => text('InputCurrency.placeholder', 'type something'),
+  errorMsg: () => text('InputCurrency.errorMsg', 'you did not type something'),
+  defaultText: () => text('InputCurrency.defaultText', '0'),
+  max: () => number('InputCurrency.max', 10000),
+  min: () => number('InputCurrency.min', -1000),
+  step: () => number('InputCurrency.step', 0.01),
+  format: () => object('InputCurrency.format', {
+    mantissa: 0,
```
Comment (smurrayatwork) [inline react/src/components/atoms/forms/InputCurrency/InputCurrency.knobs.options.js]:
```suggestion
    mantissa: 2,
```

File: react/src/components/atoms/forms/InputCurrency/index.js [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+                update.errorMsg = errorMsg;
+              }
+            } else {
+              errorMsg = '';
+              update.showError = false;
+              update.errorMsg = errorMsg;
+            }
+            context.updateState(update);
+            if (typeof props.onChange === 'function') {
+              props.onChange(numberValue);
```
Comment (smurrayatwork) [inline react/src/components/atoms/forms/InputCurrency/index.js]:
```suggestion
              props.onChange(numberValue, e);
```

File: react/src/components/atoms/forms/InputCurrency/index.js (hunk 2 above)
Comment (smurrayatwork) [reply react/src/components/atoms/forms/InputCurrency/index.js:280]: Added.

**Review by avrilpearl — APPROVED** [review]
lgtm - just review/update the button addition from me

**Review by clairesunstudio — APPROVED** [review]
looks good! We could make some improvements later on to format the default value to currency format

**Pull request merged**